### PR TITLE
New Class for Specs Documentation

### PR DESCRIPTION
**Pull request by welguisz**
This class allows for documentation to control how the map output and the topic output would look.  This is just the base class and individual companies and people can add on to make it specific to their strategy.

### CODE REVIEW

**Comment by priyavadan** [comment]
@welguisz this needs a lint fix :)

**Comment by welguisz** [comment]
Wait until my next commit.  I noticed that I left out some spec tests.

**Comment by welguisz** [comment]
This looks good to go.  Can I get a new release of Origen in the next 24 hours so I can continue on some of my projects.  Thanks!

File: lib/origen/specs.rb [hunk 1]
```diff
@@ -324,6 +332,53 @@ def doc_resources(options = {})
...
+      }.update(options)
+      return nil if @_documentation.nil?
+      return nil if @_documentation.empty?
+      doc_found = Hash.new do |h, k|
+        h[k] = Hash.new do |hh, kk|
+          hh[kk] = Hash.new do |hhh, kkk|
+            hhh[kkk] = Hash.new do |hhhh, kkkk|
+              hhhh[kkkk] = Hash.new do |hhhhh, kkkkk|
+                hhhhh[kkkkk] = Hash.new do |hhhhhh, kkkkkk|
+                  hhhhhh[kkkkkk] = {}
```
Comment (ginty) [inline lib/origen/specs.rb]: Yikes!
Don't need to change it right now, but this trick should help with this kind of thing:
https://github.com/franzejr/best-ruby/blob/master/tricks/autovivification.md

**Pull request merged**